MedallionShell/Streams/MergedLinesEnumerable.cs

```diff
@@ -92,6 +110,68 @@ private IEnumerator<string> GetEnumeratorInternal()
             }
         }
 
+#if NETSTANDARD2_0_OR_GREATER
+        private async IAsyncEnumerator<string> GetAsyncEnumeratorInternal()
+        {
+            var tasks = new List<ReaderAndTask>(capacity: 2)
+            {
+                new(this.standardOutput),
+                new(this.standardError),
+            };
+
+            // phase 1: read both streams simultaneously, alternating between which is given priority.
+            // Stop when one (or both) streams is exhausted
+
+            TextReader remaining;
+            while (true)
+            {
+                ReaderAndTask next;
+                if (tasks[0].Task.IsCompleted)
+                {
+                    next = tasks[0];
+                }
+                else if (tasks[1].Task.IsCompleted)
+                {
+                    next = tasks[1];
+                }
+                else
+                {
+                    var nextCompleted = await Task.WhenAny(tasks.Select(t => t.Task));
+                    next = tasks[0].Task == nextCompleted ? tasks[0] : tasks[1];
+                }
+
+                var nextLine = await next.Task;
+                tasks.Remove(next);
+
+                if (nextLine != null)
+                {
+                    yield return nextLine;
+                    tasks.Add(new ReaderAndTask(next.Reader));
+                }
+                else
+                {
+                    var otherAsyncLine = await tasks[0].Task;
+                    if (otherAsyncLine != null)
+                    {
+                        yield return otherAsyncLine;
+                        remaining = tasks[0].Reader;
+                        break;
+                    }
+                    else
+                    {
+                        yield break;
+                    }
+                }
+            }
+
+            // phase 2: finish reading the remaining stream
+            while (remaining.ReadLine() is { } line)
+            {
+                yield return line;
+            }
+        }
+#endif
+
         private struct ReaderAndTask : IEquatable<ReaderAndTask>
         {
             public ReaderAndTask(TextReader reader)
```

Comment:
We'll want this to now take in optional `CancellationToken`, so that in NET8_0_OR_GREATER we can call `ReadLineAsync(cancellationToken)`